HANDBOOK.md

```diff
@@ -0,0 +1,180 @@
+# The Handbook for extension creation on the Echo media player  
+Thanks for choosing to contribute to Echo by creating your own extension, we will aim to facilitate the process of creating the extension, unlike Shivam who expects you to read the source code for Echo to figure out how to write an extension, this easy-to-understand guide will give you the necessary details and steps to contribute with the less effort spent  
+> [!NOTE]  
+> You are still expected to have some knowledge in coding (kotlin) and creating Android projects  
+> [!NOTE]
+> This handbook does not try to cover the entire extension API and you should refer to javadoc for more insight, we will only walk you through the basics so that you can get the hang of things quick  
+
+## Setting up your building system  
+**For people who use Android Studio please skip this part**  
+Create a keystore by issuing the following command:  
+```sh
+keytool -genkeypair -v -keystore my-release-key.keystore -alias my-app-key -keyalg RSA -keysize 2048 -validity 10000
+```
+Make sure to save this file somewhere safe OUTSIDE THE REPO, I usually place it in `~/.keystore/`  
+Take note of the __key alias__ and the __password__  
+Save your keystore password to a file, most effectively done like so:  
+```sh
+echo "YOUR KEYSTORE PASSWORD" > pass.txt
+```
+And make sure to place it OUTSIDE THE REPO as well  
+Create a build script that contains the following:  
+```sh
+#!/bin/bash
+
+pass=$(cat "/home/nekomimi/.keystore/pass.txt")
+./gradlew assembleRelease \
+    -Pandroid.injected.signing.store.file="/home/nekomimi/.keystore/my-release-key.keystore" \
+    -Pandroid.injected.signing.store.password="$pass" \
+    -Pandroid.injected.signing.key.alias="my-app-key" \
+    -Pandroid.injected.signing.key.password="$pass"
+```
+Make sure to replace the actual location of the keystore file and password file to your own location  
+To disable debugging features when building:  
+- Open `app/build.gradle.kts`  
+- Inside the `android {}` scope, find `buildTypes`, generally you'll only find an `all` entry, inside it (`all`) add `isMinifyEnabled = true` inside the scope  
+
+## What is an extension ?  
+We can't dive straight into creating the extension without first knowing how one works, apart from the main code that sets up a kotlin/Android project you start with your main class, that class extends multiple interfaces to add the features you need (lets say you want to create an extension that only does lyrics, then you'd need the LyricsClient interface)  
+We will divide our code into segements to keep things easier to read and help others be able to contrib to your project, we will also stick to the kotlin indent style and syntax rules (because people will shit on you otherwise)  
+Your main class file should only contain declarations and overrides, you are advised to not implement any of the logic in this file but rather to call functions from other files  
```

Comment:
I recommend to replace this line with 
> Its better to have your main class file only contain declarations and overrides, you are advised to not implement any of the logic in this file but rather to call functions from other files. This will keep your main class clean and readable.